core/templates/domain/summary/exploration-summary-backend-api.service.ts

```diff
@@ -17,77 +17,90 @@
  * from the backend.
  */
 
-require('services/alerts.service.ts');
-require('services/validators.service.ts');
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-angular.module('oppia').factory('ExplorationSummaryBackendApiService', [
-  '$http', '$q', 'AlertsService',
-  'ValidatorsService', 'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, AlertsService,
-      ValidatorsService, EXPLORATION_SUMMARY_DATA_URL_TEMPLATE) {
-    var _fetchExpSummaries = function(
-        explorationIds, includePrivateExplorations, successCallback,
-        errorCallback) {
-      if (!explorationIds.every(ValidatorsService.isValidExplorationId)) {
-        AlertsService.addWarning('Please enter a valid exploration ID.');
+import { AppConstants } from 'app.constants';
+import { AlertsService } from 'services/alerts.service';
+import { ValidatorsService } from 'services/validators.service';
 
-        var returnValue = [];
-        for (var i = 0; i < explorationIds.length; i++) {
-          returnValue.push(null);
-        }
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationSummaryBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private alertsService: AlertsService,
+    private validatorsService: ValidatorsService) {}
 
-        if (errorCallback) {
-          errorCallback(returnValue);
-        }
-        return;
+  private _fetchExpSummaries(
+      explorationIds: string[],
+      includePrivateExplorations: boolean,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
```

Comment:
Don't use Object. The structure of exploration summary is this:
```
{
            'category': u'A category',
            'community_owned': False,
            'id': 'eid2',
            'language_code': 'en',
            'num_views': 0,
            'objective': u'An objective',
            'status': 'public',
            'tags': [],
            'thumbnail_bg_color': '#a33f40',
            'thumbnail_icon_url': self.get_static_asset_url(
                '/images/subjects/Lightbulb.svg'),
            'title': u'Exploration 2 Albert title',
}
 Create an interface called ExplorationSummaryBackendDict.
```

Reply:
+cc @nishantwrp.

Reply:
This structure is present in the creator exploration summary object factory and learner exploration summary object factory, which one should I use or should I create another interface?

Reply:
Done!